### ISSUE TITLE
Install a plugin using SSHFS for the plugins forlder

### ISSUE THREAD

**Issue by raphgilles** [Chocobozzz/PeerTube#5447]
### Describe the current behavior

**ENGLISH :**

I am currently trying to install the livechat plugin on my instance.
And I encounter a problem that I had already encountered for other plugins.

I'm using SSHFS to connect to the entire PeerTube "storage" folder. It also contains the plugins folder.

When I try to install a plugin, I get this error:
`error An unexpected error occurred: EIO: i/o error, symlink`

I mount my file via SSHFS with:
`-o allow_other,reconnect,ServerAliveInterval=15,ServerAliveCountMax=3,follow_symlinks,nonempty`

I also tried with the "transform_symlinks" option but nothing worked.

Do you have any idea how I could go about it?

**FRENCH :**

J'essaye actuellement d'installer le plugin livechat sur mon instance.
Et je rencontre un problème que j'avais déjà rencontré pour d'autres plugins.

J'utilise SSHFS pour me connecter à tout le dossier "storage" de PeerTube. Il contient également le dossier plugins.

Quand j'essaye d'installer un plugin, j'obtien cette erreur :
`"error An unexpected error occurred: EIO: i/o error, symlink"`

Je monte mon dossier via SSHFS avec les options :
`-o allow_other,reconnect,ServerAliveInterval=15,ServerAliveCountMax=3,follow_symlinks,nonempty`

J'ai également essayé avec l'option "transform_symlinks" mais rien n'y fait.

Vous auriez une idée de comment je pourrais m'y prendre ?

### Steps to reproduce

1. Use SSHFS to mount the storage folder
2. Try to install or re-install a plugin after deleted it

### Additional information

* PeerTube instance:
  * URL: https://peertube.stream
  * Version: 5.0.0-rc.1
  * NodeJS version: 16.8.0
  * SSHFS version :  2.10.0
  * Ffmpeg version: Not important

**Comment by raphgilles** [comment]
Update : It seems I'm not alone.
https://serverfault.com/questions/1097292/filesystem-mounted-with-sshfs-throws-error-inside-of-docker-when-creating-a-syml

**Comment by JohnXLivingston** [comment]
Sorry, I'm a little bit busy right now. Did not read in detail.
Are you in a Docker container?

If so, it is probably because of Docker. Which don't allow FUSE (unless you are in a privileged container, but... that's not good for security).
Is your sshfs inside or outside your container?

I'm not a Docker/FUSE/SSHFS expert... I just know I had to struggle with docker when I wanted to do such kind of things (symlinks and so on).
Would it be ok for you to replace your Docker installation by a standard installation?

**Comment by JohnXLivingston** [comment]
Anyway, perhaps you will find a solution searching for something like «howto docker volume sshfs»

For example:
https://serverfault.com/questions/947182/mount-a-sshfs-volume-into-a-docker-instance
(did not read entirely)

**Comment by raphgilles** [comment]
> 

No. I'm on a standard installation without Docker. I think the issue of the link above is not due specifically to Docker but to SSHFS. Like me. The error is exactly the same.

**Comment by JohnXLivingston** [comment]
Have you errors in your peertube logs? (the easiest way to check, is using the admin web interface , where you can filter by error level)

Might be when `yarn` tries to install plugins in `storage/plugins`.

Here is how Peertube add plugins and themes (assuming your peertube is in standards folders):

Step1: It add the name and version of the plugin/theme in /var/www/peertube/storage/plugins/package.json

You should see something like this in this file:

```json
{
  "dependencies": {
    "peertube-plugin-livechat": "6.0.0-beta.1"
}
```

If not, the issue must happen before.


Step2: Peertube calls yarn to install the plugin and its dependencies

After this step, you should have a folder `/var/www/peertube/storage/plugins/node_modules/peertube-plugin-livechat/` containing the plugin (and other folder beside with plugins dependencies).


Depending on which step of this process fails, we can then try different solutions.

**Comment by Chocobozzz** [comment]
Hello,

I'm sorry but we don't have enough resources to support custom FS mount, in particular network mounts.
